lib/mock-module/mock-module.ts

```diff
@@ -54,10 +54,31 @@ const mockProvider = (provider: any): Provider | undefined => {
     return undefined;
   }
 
+  // mocking all methods / properties of a class / object.
+  let value: any;
+  if (typeof provide === 'function') {
```

Comment:
in the next PR with the option to set mocking provider from outside I'll mote this part to MockService.